geoengine/colorizer.py

```diff
@@ -147,103 +213,88 @@ def from_response(response: api.Colorizer) -> "Colorizer":
         raise TypeError(f"Unknown colorizer type: {response['type']}")
 
 
+@dataclass
 class LinearGradientColorizer(Colorizer):
     '''A linear gradient colorizer.'''
     breakpoints: List[ColorBreakpoint]
-
-    def __init__(
-        self,
-        breakpoints: List[ColorBreakpoint],
-        no_data_color: Tuple[int, int, int, int],
-        default_color: Tuple[int, int, int, int]
-    ):
-        super().__init__(no_data_color, default_color, "linearGradient")
-        self.breakpoints = breakpoints
+    over_color: Rgba
+    under_color: Rgba
 
     @staticmethod
-    def from_response_linear(response: api.LinearGradientColorizer) -> "LinearGradientColorizer":
+    def from_response_linear(response: api.LinearGradientColorizer) -> LinearGradientColorizer:
         """Create a colorizer from a response."""
         breakpoints = [ColorBreakpoint.from_response(breakpoint) for breakpoint in response['breakpoints']]
         return LinearGradientColorizer(
+            type='linearGradient',
```

Comment:
i think you don't need to store the type. If the class has its own "to_api" method you can just set the type matching the class.

Reply:
Ok, removed it.

25ae408